### PR TITLE
Bump JUnit plugin from 1.29 to 1.48

### PR DESCRIPTION
**Pull request by basil**
Bumps [JUnit plugin](https://github.com/jenkinsci/junit-plugin) from 1.29 to 1.48. Subsumes #5173, fixing the `requireUpperBoundDeps` errors.

### CODE REVIEW

**Review by timja — APPROVED** [review]

**Review by res0nance — APPROVED** [review]

**Comment by timja** [comment]
This PR is now ready for merge, after ~24 hours, we will merge it if there's no negative feedback.

Thanks!

**Pull request merged**
